src/liblesma/Typecheck/Typechecker.cpp

```
@@ -1683,13 +1756,288 @@ auto Typechecker::visit(const Expression* /*node*/) -> void {}
+  if (!mainFilePath.empty() && isStdlibSourcePath(mainFilePath)) {
+    return;
+  }
+  warningDiagnostics->push_back(
+      AnalysisDiagnostic{.message = std::move(message),
+                         .span = span,
+                         .severity = AnalysisDiagnosticSeverity::Warning});
+}
+
+void Typechecker::recoverFromTypeError(const TypeCheckError& err) {
+  if (warningDiagnostics == nullptr) {
+    throw;
```

Cppcheck (2.20.0)

[error] Rethrowing current exception with 'throw;', it seems there is no current exception to rethrow. If there is no current exception this calls std

(rethrowNoCurrentException)